deta-lib/query.rkt

```diff
@@ -420,14 +420,22 @@
 
   (define-syntax-class q-order-pair
     #:literals (unquote)
-    (pattern [column:q-expr (~or (~optional (~and #:asc  dir-asc ))
-                                 (~optional (~and #:desc dir-desc))
-                                 (~optional (unquote dir-expr:expr)))]
-             #:with dir (cond
-                          [(attribute dir-expr) #'dir-expr]
-                          [(attribute dir-desc) #''desc]
-                          [else #''asc])
-             #:with e #'(cons column.e dir))))
+    (pattern [column:q-expr (~optional (~or (~and #:asc  dir-asc )
+                                            (~and #:desc dir-desc)
+                                            (unquote dir-expr:expr)))
+                            (~optional (~or (~and #:nulls-first nulls-first*)
+                                            (~and #:nulls-last  nulls-last* )
+                                            (unquote nulls-expr:expr)))]
```

Comment:
How about instead of 

```racket
(~optional ...) 
(~optional ...)
```

, we use 

```racket
(~optional 
 (~seq 
  (~or asc desc ...) 
  (~optional (~or nulls ...)))
```
? That way it will be a syntax error to provide a null direction w/o explicitly providing a regular direction, and it'll make the dynamic case less ambiguous.

Reply:
Fully agree! Pushed.